spec/controllers/slack_controller_spec.rb

```diff
@@ -0,0 +1,197 @@
+require 'rails_helper'
+
+describe ::DiscourseSlack::SlackController do
+  routes { ::DiscourseSlack::Engine.routes }
+
+  PLUGIN_NAME = ::DiscourseSlack.plugin_name.freeze
+
+  def get_value(key)
+    PluginStore.get(PLUGIN_NAME, key)
+  end
+
+  before do
+    SiteSetting.slack_outbound_webhook_url = "https://hooks.slack.com/services/abcde"
+    SiteSetting.slack_enabled = true
+  end
+
+  context "when logged in" do
+    let!(:user) { log_in(:admin) }
+
+    it "checking existence of default filters" do
+      expect(PluginStoreRow.where(plugin_name: PLUGIN_NAME).count).to eq(3)
+      expect(get_value("not_first_time")).to eq(true)
+    end
+
+    context '#index' do
+      it "returns a list of filters" do
+        xhr :get, :list
+        expect(response).to be_success
+        json = ::JSON.parse(response.body)
+        filters = json['slack']
+        expect(filters.count).to eq(1)
+        expect(filters[0]).to include(
+          "channel" => "#general",
+          "category_id" => "*",
+          "tags" => [],
+          "filter" => "follow"
+        )
+      end
+    end
+
+    context '#create' do
+      it "creates a filter" do
+        expect {
+          channel, category_id, filter = "#hello", "1", "follow"
+          xhr :post, :edit, { channel: channel, category_id: category_id, filter: filter }
+          expect(response).to be_success
+          id = get_value("_category_#{category_id}_#{channel}")
+          data = get_value("filter_#{id}")
+          expect(data).to include(
+            "channel" => channel,
+            "category_id" => category_id,
+            "filter" => filter
+          )
+        }.to change(PluginStoreRow, :count).by(2)
```

Comment:
Checking the count is not enough because you're not checking if the record with the right attributes is actually being created. A record with different attributes may be created as a result of a bug and this test will still pass.